src/main/scala/com/softwaremill/realworld/users/UsersRepository.scala

```diff
@@ -49,18 +49,29 @@ class UsersRepository(quill: Quill.Sqlite[SnakeCase]):
   ).unit
     .pipe(mapUniqueConstraintViolationError)
 
-  def updateByEmail(updateData: UserUpdateData, email: String): Task[UserUpdateData] = run(
-    queryUser
+  def updateByEmail(updateData: UserUpdateData, email: String): IO[Throwable, Option[UserData]] = {
```

Comment:
**question:** Can we have a narrower exception scope that Throwable here? What are actual possible errors?

Reply:
I think that we can't, because this transaction code `transaction {...}` return ZIO[*, Throwable, *], so if the return type of the updateByEmail was narrower e.g. Exception, an exception would be thrown. So we would have to change transaction on something different if we wanted to change the return type.